llvm/test/CodeGen/AMDGPU/amdhsa-kernarg-preload-num-sgprs.ll

```diff
@@ -0,0 +1,14 @@
+; RUN: llc -mtriple=amdgcn-amd-amdhsa -mcpu=gfx940 -filetype=obj < %s > %t
+; RUN: llvm-objdump -s -j .rodata %t | FileCheck --check-prefix=OBJDUMP %s
+
+; OBJDUMP: Contents of section .rodata:
+; OBJDUMP-NEXT: 0000 00000000 00000000 10010000 00000000
+; OBJDUMP-NEXT: 0010 00000000 00000000 00000000 00000000
+; OBJDUMP-NEXT: 0020 00000000 00000000 00000000 00000000
+; OBJDUMP-NEXT: 0030 4000af00 94130000 1a000400 00000000
+; OBJDUMP-NOT: 0030 0000af00 94130000 1a000400 00000000
```

Comment:
We need a human readable asm output for reference, and should check the different kind of SGPR usage numbers

Reply:
> We need a human readable asm output for reference, and should check the different kind of SGPR usage numbers

The annoying part about this bug is we don't directly output a directive for this field anywhere, it's totally derivative. To find the problem I was modifying the KD in the binary directly.

Reply:
So we cannot go from `.ll` -> `.s` -> `.o`? (i.e., will `AMDGPUAsmParser`'s calculation of SGPRBlocks still be correct?)

Reply:
I think we should be changing the reported used register count, not just manipulating the encoding fields